test/integration/daemonset/daemonset_test.go

```diff
@@ -310,112 +436,179 @@ func forEachStrategy(t *testing.T, tf func(t *testing.T, strategy *apps.DaemonSe
+
+			ds := newDaemonSet("foo", ns.Name)
+			ds.Spec.UpdateStrategy = *strategy
+			_, err := dsClient.Create(ds)
+			if err != nil {
+				t.Fatalf("Failed to create DaemonSet: %v", err)
+			}
 
-		addNodes(nodeClient, 0, 5, nil, t)
+			defer cleanupDaemonSets(t, clientset, ds)
+
+			node := newNode("single-node", nil)
+			node.Status.Conditions = []v1.NodeCondition{
+				{Type: v1.NodeReady, Status: v1.ConditionFalse},
+			}
+			_, err = nodeClient.Create(node)
+			if err != nil {
+				t.Fatalf("Failed to create node: %v", err)
+			}
 
-		validateDaemonSetPodsAndMarkReady(podClient, podInformer, 5, t)
-		validateDaemonSetStatus(dsClient, ds.Name, 5, t)
+			validateDaemonSetPodsAndMarkReady(podClient, podInformer, 1, t)
+			validateDaemonSetStatus(dsClient, ds.Name, 1, t)
+		})
 	})
 }
 
-func TestNotReadyNodeDaemonDoesLaunchPod(t *testing.T) {
+func TestInsufficientCapacityNodeDaemonDoesNotLaunchPod(t *testing.T) {
 	forEachStrategy(t, func(t *testing.T, strategy *apps.DaemonSetUpdateStrategy) {
 		server, closeFn, dc, informers, clientset := setup(t)
 		defer closeFn()
-		ns := framework.CreateTestingNamespace("simple-daemonset-test", server, t)
+		ns := framework.CreateTestingNamespace("insufficient-capacity", server, t)
 		defer framework.DeleteTestingNamespace(ns, server, t)
 
 		dsClient := clientset.AppsV1().DaemonSets(ns.Name)
-		podClient := clientset.CoreV1().Pods(ns.Name)
 		nodeClient := clientset.CoreV1().Nodes()
-		podInformer := informers.Core().V1().Pods().Informer()
+		eventClient := clientset.CoreV1().Events(ns.Namespace)
 
 		stopCh := make(chan struct{})
+		defer close(stopCh)
+
 		informers.Start(stopCh)
 		go dc.Run(5, stopCh)
-		defer close(stopCh)
 
 		ds := newDaemonSet("foo", ns.Name)
+		ds.Spec.Template.Spec = resourcePodSpec("node-with-limited-memory", "120M", "75m")
 		ds.Spec.UpdateStrategy = *strategy
 		_, err := dsClient.Create(ds)
 		if err != nil {
 			t.Fatalf("Failed to create DaemonSet: %v", err)
 		}
 		defer cleanupDaemonSets(t, clientset, ds)
 
-		node := newNode("single-node", nil)
-		node.Status.Conditions = []v1.NodeCondition{
-			{Type: v1.NodeReady, Status: v1.ConditionFalse},
-		}
+		node := newNode("node-with-limited-memory", nil)
+		node.Status.Allocatable = allocatableResources("100M", "200m")
 		_, err = nodeClient.Create(node)
 		if err != nil {
 			t.Fatalf("Failed to create node: %v", err)
 		}
 
-		validateDaemonSetPodsAndMarkReady(podClient, podInformer, 1, t)
-		validateDaemonSetStatus(dsClient, ds.Name, 1, t)
+		validateFailedPlacementEvent(eventClient, t)
 	})
 }
 
-func TestInsufficientCapacityNodeDaemonDoesNotLaunchPod(t *testing.T) {
+// TestInsufficientCapacityNodeDaemonSetCreateButNotLaunchPod tests that the DaemonSet will
+// create Pods for the node with insufficient resource, but kube-scheduler did not launch it
+// when ScheduleDaemonSetPods is enabled.
+func TestInsufficientCapacityNodeWhenScheduleDaemonSetPodsEnabled(t *testing.T) {
```

Comment:
Enabling this feature will break some DaemonSet unit tests, such as `TestInsufficientCapacityNodeDaemonDoesNotLaunchPod`